src/ja/common/docker_context.py

```diff
@@ -21,18 +21,38 @@ def __init__(self, source_path: str, mount_path: str):
         @param source_path The host directory to be mounted.
         @param mount_path The target path in the container to mount at.
         """
+        self._source_path = source_path
+        self._mount_path = mount_path
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, MountPoint):
+            return self.source_path == other.source_path and self.mount_path == other.mount_path
+        else:
+            return False
 
     @property
     def source_path(self) -> str:
         """!
         @return The directory to be mounted.
         """
+        return self._source_path
 
     @property
     def mount_path(self) -> str:
         """!
         @return The path where the directory should be mounted.
         """
+        return self._mount_path
+
+    def to_dict(self) -> Dict[str, object]:
```

Comment:
`return self.__dict__ `is enough for all the to_dict methods. Only difference is that the names of the attributes in the dictionary are going to start with an underscore.

Reply:
No it's not.
`self.__dict__` will directly return _all_ attributes set for an object, including attributes that are not relevant for the state of the object.
You would also still have to convert any non-primitive attribute into a dictionary so you could only use this implementation for the most simple of objects.